pkg/util/yaml/decoder.go

```diff
@@ -52,10 +53,16 @@ type YAMLToJSONDecoder struct {
 // stream in chunks by converting each document (as defined by
 // the YAML spec) into its own chunk, converting it to JSON via
 // yaml.YAMLToJSON, and then passing it to json.Decoder.
-func NewYAMLToJSONDecoder(r io.Reader) *YAMLToJSONDecoder {
+func NewYAMLToJSONDecoder(r io.Reader, bytes []byte, isStdin bool) *YAMLToJSONDecoder {
 	reader := bufio.NewReader(r)
-	return &YAMLToJSONDecoder{
-		reader: NewYAMLReader(reader),
+	if isStdin && sessionIsTTY() {
+		return &YAMLToJSONDecoder{
+			reader: NewYAMLByteReader(bytes),
```

Comment:
Good question. Maybe I could try to do `Reader.ReadRune()` instead of `Peek()`. We could guess if it's a JSON stream and then I could call `Reader.UnreadRune()` to place back the cursor at the beginning.

I'll give it a shot tonight